test/sql/function/string/test_damerau_levenshtein.test

```diff
@@ -0,0 +1,341 @@
+
+query I
+SELECT damerau_levenshtein('great', '')
+----
+5
+
+query I
+SELECT damerau_levenshtein('', '')
+----
+0
+
+query I
+SELECT damerau_levenshtein(NULL, 'drive')
+----
+NULL
+
+query I
+SELECT damerau_levenshtein('drive', NULL)
+----
+NULL
+
+query I
+SELECT damerau_levenshtein(NULL, NULL)
+----
+NULL
+
+query I
+SELECT damerau_levenshtein('', NULL)
+----
+NULL
+
+query I
+SELECT damerau_levenshtein(NULL, '')
+----
+NULL
+
+
+statement error
+SELECT damerau_levenshtein('one', 'two', 'three')
+
+statement error
+SELECT damerau_levenshtein('one')
+
+statement error
+SELECT damerau_levenshtein()
+
+
+statement ok
+CREATE TABLE strings(s VARCHAR)
+
+statement ok
+INSERT INTO strings VALUES ('here'), ('heres'), ('there'), ('three'), ('threes')
+
+query I
+SELECT damerau_levenshtein(s, 'theres') FROM strings ORDER BY s
+----
+2
+1
+1
+2
+1
+
+query I
+SELECT damerau_levenshtein('herse', s) FROM strings ORDER BY s
+----
+1
+1
+2
+3
+3
+
+query I
+SELECT damerau_levenshtein(NULL, s) FROM strings
+----
+NULL
+NULL
+NULL
+NULL
+NULL
+
+query I
+SELECT damerau_levenshtein(NULL, s) FROM strings
+----
+NULL
+NULL
+NULL
+NULL
+NULL
+
+query I
+SELECT damerau_levenshtein('', s) FROM strings ORDER BY s
+----
+4
+5
+5
+5
+6
+
+query I
+SELECT damerau_levenshtein(s, '') FROM strings ORDER BY s
+----
+4
+5
+5
+5
+6
+
+
+statement ok
+DROP TABLE strings
+
+statement ok
+CREATE TABLE strings(s VARCHAR)
+
+statement ok
+INSERT INTO strings VALUES (NULL)
+
+query I
+SELECT damerau_levenshtein(s, NULL) from strings
+----
+NULL
+
+query I
+SELECT damerau_levenshtein(NULL, s) from strings
+----
+NULL
+
+query I
+SELECT damerau_levenshtein('test', s) from strings
+----
+NULL
+
+query I
+SELECT damerau_levenshtein(s, 'test') from strings
+----
+NULL
+
+query I
+SELECT damerau_levenshtein('null', s) from strings
+----
+NULL
+
+query I
+SELECT damerau_levenshtein('', s)  FROM strings
+----
+NULL
+
+query I
+SELECT damerau_levenshtein(s, '')  FROM strings
+----
+NULL
+
+
+statement ok
+DROP TABLE strings
+
+statement ok
+CREATE TABLE strings(s VARCHAR)
+
+statement ok
+INSERT INTO strings VALUES ('')
+
+query I
+SELECT damerau_levenshtein(NULL, s)  FROM strings
+----
+NULL
+
+query I
+SELECT damerau_levenshtein(s, NULL)  FROM strings
+----
+NULL
+
+query I
+SELECT damerau_levenshtein(s, '')  FROM strings
+----
+0
+
+query I
+SELECT damerau_levenshtein('', s)  FROM strings
+----
+0
+
+query I
+SELECT damerau_levenshtein(s, 'test')  FROM strings
+----
+4
+
+query I
+SELECT damerau_levenshtein('test', s)  FROM strings
+----
+4
+
+query I
+SELECT damerau_levenshtein('null', s)  FROM strings
+----
+4
+
+
+statement ok
+DROP TABLE strings
+
+statement ok
+CREATE TABLE strings(s_left VARCHAR, s_right VARCHAR)
+
+statement ok
+INSERT INTO strings VALUES 	('identical', 'identical'), ('dientical', 'identical'),
+							('dinetcila', 'identical'), ('abcdefghijk', 'bacdfzzeghki'),
+							('abcd', 'bcda'), ('great', 'greta'),
+							('abcdefghijklmnopqrstuvwxyz', 'abdcpoxwz'),
```

Comment:
I want to say we should add some extra tests for strings > 12 characters, because our `string_t` inlines strings up to 12 characters, so memory issues don't really become apparent unless longer strings are used.
But I don't think it matters here, you're not creating strings, only reading existing ones.